### PR TITLE
Allow Texture static create functions (not Parse) to take Engine

### PR DESCRIPTION
**Pull request by Palmer-JC**

### CODE REVIEW

File: packages/dev/core/src/Materials/Textures/cubeTexture.ts [hunk 1]
```diff
@@ -119,34 +119,38 @@ export class CubeTexture extends BaseTexture {
...
      * @returns the prefiltered texture
      */
-    public static CreateFromPrefilteredData(url: string, scene: Scene, forcedExtension: any = null, createPolynomials: boolean = true) {
-        const oldValue = scene.useDelayedTextureLoading;
-        scene.useDelayedTextureLoading = false;
-
-        const result = new CubeTexture(url, scene, null, false, null, null, null, undefined, true, forcedExtension, createPolynomials);
+    public static CreateFromPrefilteredData(url: string, sceneOrEngine: Scene | ThinEngine, forcedExtension: any = null, createPolynomials: boolean = true) {
+        let oldValue = false;
+        if (sceneOrEngine instanceof Scene) {
```
Comment (sebavan) [inline packages/dev/core/src/Materials/Textures/cubeTexture.ts:144]: As you can not import the Scene Value, I would advise to only check for (sceneOrEngine as Scene).useDelayedTextureLoading !== "undefined)

File: packages/dev/core/src/Materials/Textures/cubeTexture.ts [hunk 2]
```diff
@@ -119,34 +119,38 @@ export class CubeTexture extends BaseTexture {
...
+            oldValue = sceneOrEngine.useDelayedTextureLoading;
+            sceneOrEngine.useDelayedTextureLoading = false;
+        }
 
-        scene.useDelayedTextureLoading = oldValue;
+        const result = new CubeTexture(url, sceneOrEngine, null, false, null, null, null, undefined, true, forcedExtension, createPolynomials);
 
+        if (sceneOrEngine instanceof Scene) {
```
Comment (sebavan) [inline packages/dev/core/src/Materials/Textures/cubeTexture.ts:151]: Instead of doing the check back, you could store the result of the test in a variable

**Review by Popov72 — CHANGES_REQUESTED** [review]

**Comment by Popov72** [comment]
`_IsScene` is private to `BaseTexture`, that's why you can't access it. Actually, it is simply doing `return sceneOrEngine.getClassName() === "Scene"`, that's why I suggested to use the same test.

**Comment by sebavan** [comment]
If you follow sceneOrEngine.getClassName() === "Scene" you won t need to remove the type at the top :-)

**Comment by Popov72** [comment]
What do you mean?

It seems testing `sceneOrEngine.getClassName() === "Scene"` will work with `import type { Scene } from "../../scene";`

**Comment by sebavan** [comment]
Yup, exactly that :-) no need to do this change https://github.com/BabylonJS/Babylon.js/pull/13221/files#diff-03f2af9a6752fe469eb572ec0afa30d2ff5ae943f0a1e1c59020cc54bd7c1264R4

**Comment by Palmer-JC** [comment]
Am a little lost here, but something is working right off the current CDN.babylon.js.  I thought I had switched the framework to the one I had made, but I am only using the resulting D.TS in the transpile.  Yet it runs.

After that, I double checked (right from the console this time), and engine is in the call (and the call that calls it).
```javascript
(function (XR_UI) {
    class Environment {
        static Get(engine) {
            if (!Environment._Cube) {
                Environment._Cube = BABYLON.CubeTexture.CreateFromPrefilteredData(Environment._Url, engine, '.env');
                Environment._Cube.name = null;
                Environment._Url = null;
            }
            return Environment._Cube;
        }
    }
    Environment._Url = 'data:application/...
```

So, is the only thing needed is the D.TS file????

**Comment by sebavan** [comment]
It is expected as you basically only changed typings in the PR and preventing setting useDelayedTextureLoading and In JS setting a field prevently undefined would basically do not break. The only thing is engine would have a property useDelayedTextureLoading :-)

**Comment by Palmer-JC** [comment]
I finally got the whole picture, in that "I don't need no stinking CreateFromPrefilteredData()".  It is only just a 4 line static function of which in my case, I only need one, the public constructor.

I just added the constructor in the little class with the embedded data, and I am done.  Sorry for your trouble.

**Pull request closed without merging**